파티찾기 게시글 등록 기능 구현

Minggurazi opened the pull request:
## 개요

- 파티찾기 게시글 등록 기능 구현

<br>

## 주요 변경 내역

- 게시글 등록 기능 구현

<br>

## Issue 번호 및 관련 링크

- closes #39

Minggurazi committed:
feat : 파티찾기 게시글 등록 기능 구현

Minggurazi committed:
feat : 음성채팅 여부 누락 추가

Minggurazi committed:
feat : 여백 추가

PMtHk reviewed (commented):
카드 창이 나온 상황에서 무언가를 입력했을 때,
뒤로가기 버튼 누르면 내용이 사라진다는 알림 정도 띄워주면 어떰?

Minggurazi committed:
feat : 디자인 피드백 반영

Minggurazi committed:
feat : 세부 기능 구현 추가

Minggurazi committed:
feat : 모달창에서 뒤로가기시 state값 초기화 오류 수정

Minggurazi committed:
fix : type 선택 오류 수정

Minggurazi committed:
fix : Modal UI 수정

PMtHk reviewed (changes requested):
기본적으로 우리가 백엔드와 협의된 내용은
1. 자유랭크를 위해 포지션은 배열로 받을 수 있게 한다.
2. 티어는 배열이 아닌 단일 값이다.
인데, 현재 듀오랭크는 큰 문제 없어 보이는데 다른 건 애매한 동작이 있는 것 같아.

우선, 자유랭크에서 티어를 여러개 받을 수 있도록 하는 게 필요해 보이지만, 너무 복잡해지고, 오히려 사용자의 편의를 방해할 가능성이 있어, 그래서 그냥 글 작성자가 원하는 메인 티어로 생각하고 하나로 받는 것이 나아.
그리고, 포지션에 상관없음이 있으니까, 상관없음 버튼을 누르면, 기존의 누른 버튼들은 초기화 해줄 필요가 있어, 예를 들어, 탑 정글을 고른 상태에서, 상관없음을 추가로 누르면, 배열을 비우고, ALL을 추가하는 방식으로.

일반도, 사실상 자유랭크의 티어를 뺀 느낌이 강하니까, 위와 동일한 방식이지.
칼바람나락(무작위 총력전)의 경우는 사실상 티어, 포지션이 의미가 없으니까 아래 버튼들도 필요가 없어서 그냥 disabled 처리하면 될 것 같아.

정리하면, 
- 어느 타입이든 티어는 하나를 정하도록 할 것(상관없음을 넣을 거면 넣어도 됨).
- 포지션을 배열로 저장하지만, 상관없음이 눌리면 배열에 ALL 하나만 들어가도록 할 것.
- 무작위 총력전을 선택하면, 아래 티어와 포지션은 disabled 처리할 것.

최대한 이해하기 쉽게 적으려고 했는데, 이해 안되면 카톡 남겨놓으면 시간 될때 설명해줄게.

Minggurazi committed:
fix : ToggleButton value값 단일 select로 수정

Minggurazi committed:
fix : 코드 정리

PMtHk committed:
style : 게시글 작성 모달 디자인 수정

- 모달 디자인 수정
- 사용자 입력 발생시에만 종료시 confirm 하도록 수정
- 사용자에 연결된 계정 있을시에만 연결된 계정 이용가능 하도록 수정

PMtHk committed:
style : 토글버튼 여백 수정

PMtHk committed:
지속시간 텍스트 컬러 수정 및 마이크 버튼 위치 조정

PMtHk committed:
feat : 마이크 여부 토글버튼화
